components/director/internal/api/token_resolver_test.go

```diff
@@ -0,0 +1,93 @@
+package api
+
+import (
+	"context"
+	"testing"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql/internalschema"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+
+	"github.com/stretchr/testify/mock"
+
+	"github.com/kyma-incubator/compass/components/director/internal/tokens"
+
+	"github.com/stretchr/testify/assert"
+
+	apiMocks "github.com/kyma-incubator/compass/components/director/internal/api/automock"
+	persistenceMocks "github.com/kyma-incubator/compass/components/director/pkg/persistence/automock"
+	"github.com/pkg/errors"
+)
+
+func TestTokenResolver_GenerateCSRToken(t *testing.T) {
+
+	const (
+		authId     = "authID"
+		tokenValue = "tokenValue"
+	)
+
+	t.Run("fails when transaction fails to begin", func(t *testing.T) {
+		// GIVEN
+		transactioner := &persistenceMocks.Transactioner{}
```

Comment:
Reuse the "pre-defined" mocks here - https://github.com/kyma-incubator/compass/blob/master/components/director/pkg/persistence/txtest/txtest.go

Example usage:
```
mockedTx, mockedTransactioner := txtest.NewTransactionContextGenerator(mockedError()).ThatFailsOnBegin()
defer mockedTx.AssertExpectations(t)
defer mockedTransactioner.AssertExpectations(t)
```

(this applies to the other tests here as well)